cmake/Dependencies.cmake

```diff
@@ -53,16 +58,20 @@ if (BUILD_NVJPEG2K)
 endif (BUILD_NVJPEG2K)
 
 # NVIDIA NPP library
-CUDA_find_library(CUDA_nppicc_static_LIBRARY nppicc_static)
-CUDA_find_library(CUDA_nppc_static_LIBRARY nppc_static)
-list(APPEND DALI_LIBS ${CUDA_nppicc_static_LIBRARY})
-list(APPEND DALI_EXCLUDES libnppicc_static.a)
-list(APPEND DALI_LIBS ${CUDA_nppc_static_LIBRARY})
-list(APPEND DALI_EXCLUDES libnppc_static.a)
+if (NOT WITH_DYNAMIC_CUDA_LIBS)
+  CUDA_find_library(CUDA_nppicc_LIBRARY nppicc_static)
+  CUDA_find_library(CUDA_nppc_LIBRARY nppc_static)
+  list(APPEND DALI_LIBS ${CUDA_nppicc_LIBRARY})
+  list(APPEND DALI_EXCLUDES libnppicc_static.a)
+  list(APPEND DALI_LIBS ${CUDA_nppc_LIBRARY})
+  list(APPEND DALI_EXCLUDES libnppc_static.a)
+endif (NOT WITH_DYNAMIC_CUDA_LIBS)
 
 # cuFFT library
-CUDA_find_library(CUDA_cufft_static_LIBRARY cufft_static)
-list(APPEND DALI_EXCLUDES libcufft_static.a)
+if (NOT WITH_DYNAMIC_CUDA_LIBS)
+  CUDA_find_library(CUDA_cufft_LIBRARY cufft_static)
+  list(APPEND DALI_EXCLUDES libcufft_static.a)
+endif (NOT WITH_DYNAMIC_CUDA_LIBS)
 
 # CULIBOS needed when using static CUDA libs
```

Comment:
So is it needed in dynamic linking case?

Reply:
Good catch. I need to check it.

Reply:
... on the other head dlopen doesn't load other libs so we would need to load them on our own and provide the dynamic wrapper as well. I'm not sure if we want to maintain such entity we don't use directly.
@mzient - what do you think?

Reply:
It seems to work, and dynamic libs doesn't require CULIBOS. Let me run the CI to be sure.